Core/src/org/sleuthkit/autopsy/contentviewers/PersonaAccountFetcher.java

```diff
@@ -22,116 +22,138 @@
-                    // No persona found
-                    personaLabelText += Bundle.CommunicationArtifactViewerHelper_persona_unknown();
-
-                    // show a 'Create' button
-                    personaButtonText = Bundle.CommunicationArtifactViewerHelper_persona_button_create();
-                    buttonActionListener = new CreatePersonaButtonListener(parentComponent, personaSearcherData);
-                } else {
-                    Persona persona = personas.iterator().next();
-                    personaLabelText += persona.getName();
-                    if (personas.size() > 1) {
-                        personaLabelText += Bundle.PersonaDisplayTask_persona_count_suffix(Integer.toString(personas.size()));
-                    }
-                    // Show a 'View' button
-                    personaButtonText = Bundle.CommunicationArtifactViewerHelper_persona_button_view();
-                    buttonActionListener = new ViewPersonaButtonListener(parentComponent, persona);
-                }
+        try {
+            Map<String, Collection<Persona>> accountMap = get();
 
-                personaSearcherData.getPersonaNameLabel().setText(personaLabelText);
-                personaSearcherData.getPersonaActionButton().setText(personaButtonText);
-                personaSearcherData.getPersonaActionButton().setEnabled(true);
-
-                // set button action
-                personaSearcherData.getPersonaActionButton().addActionListener(buttonActionListener);
-            } catch (CancellationException ex) {
-                logger.log(Level.INFO, "Persona searching was canceled."); //NON-NLS
-            } catch (InterruptedException ex) {
-                logger.log(Level.INFO, "Persona searching was interrupted."); //NON-NLS
-            } catch (ExecutionException ex) {
-                logger.log(Level.SEVERE, "Fatal error during Persona search.", ex); //NON-NLS
+            for (AccountPersonaSearcherData searcherData : personaSearchDataList) {
+                Collection<Persona> persona = accountMap.get(searcherData.getAccountIdentifer());
+                updateDataObject(searcherData, persona);
             }
 
+        } catch (CancellationException ex) {
+            logger.log(Level.INFO, "Persona searching was canceled."); //NON-NLS
+        } catch (InterruptedException ex) {
+            logger.log(Level.INFO, "Persona searching was interrupted."); //NON-NLS
+        } catch (ExecutionException ex) {
+            logger.log(Level.SEVERE, "Fatal error during Persona search.", ex); //NON-NLS
         }
-        
+
+        parentComponent.repaint();
+    }
+
+    /**
+     * Update the Persona gui controls.
+     *
+     * @param personaSearcherData The data objects with persona controls
+     * @param personas            Collection of persona objects
+     */
+    private void updateDataObject(AccountPersonaSearcherData personaSearcherData, Collection<Persona> personas) {
```

Comment:
I'd recommend to change the name of this method, updateDataObject is very vague as to what its updating.

maybe updatePersonaControls() or some such.